core/src/test/java/org/apache/iceberg/actions/TestBinPackStrategy.java

```diff
@@ -187,10 +212,10 @@ public void testGroupingWithDeletes() {
   @Test
   public void testMaxGroupSize() {
     RewriteStrategy strategy = defaultBinPack().options(ImmutableMap.of(
-        RewriteDataFiles.MAX_FILE_GROUP_SIZE_BYTES, Long.toString(1000 * MB)
+        RewriteDataFiles.MAX_FILE_GROUP_SIZE_BYTES, Long.toString(1300 * MB)
```

Comment:
Why is this test changed?

Reply:
Hmmm this makes me think that maybe I was wrong to change that parameter. I think we want groups in in any of the following conditions

1. Any group that has more than min_input_files
2. Any group which will successfully write a brand new file closer to target file size
3. Any group which contains a file larger than max_target file size
4. Any group which has more deletes than the dele threshold

I think 2 is sizeOfInput(files) > target file size and numFiles > 1 
I think 3 is sizeOfInput(files) > maxTargetFileSize

Reply:
For 1 if we just allow `Any group that has more than min_input_files` then when min_input_files = 1 we will plan it irrespective if the sizes were already within threshold .

+1 on 2, 3

Reply:
Yep for 1) we can add in files > 1,